### PR TITLE
Students program. Angular JS. Task 2

### PR DESCRIPTION
**Pull request by digitalmaxing**

### CODE REVIEW

File: script/controller/tasklist-front.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    .controller('widgetController', ['$scope', '$state', 'Tasks', 'Utils', function ($scope, $state, Tasks, Utils) {
+
+        $scope.description = "";
+        $scope.assignee = "";
+        $scope.onlyAssigned = $state.is('tasks.assigned');
+
+        $scope.items = Tasks.getItems();
+
+        $scope.items.forEach(function (item) {
+            item.hovered = false;
```
Comment (natalia-zolochevska) [inline script/controller/tasklist-front.js:16]: We discussed that on a lecture: model should not be aware of ui-related stuff.
selected is OK, but hovered and focused is too ui-specific

File: views/task-items.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+         ng-mouseleave="item.hovered = false;">
+        <form name="taskItemForm" novalidate>
+            <div class="task-properties">
+                {{item.task.creationDate | datetime }}
+                <b>Assigned to:</b>
+                <input name="assignee" ng-show="item.access.edit" class='assign inline-edit' placeholder="Assign to"
+                       ng-model="item.task.assignee" required>
+                <input type="submit" class="btn assign-btn" ng-click="reassignTask(item)" value="Assign"
+                       ng-show="item.access.edit" ng-disabled="taskItemForm.assignee.$invalid">
+            </div>
```
Comment (natalia-zolochevska) [inline views/task-items.html:19]: I pointed that css is preferable way. Here you need to duplicate item id ng-disabled="taskItemForm.assignee..."
ng-disabled="taskItemForm.description..."
this overloads code and makes it unreadable. 
with css you can handle all fields with single piece of code

File: script/controller/tasklist-front.js (hunk 1 above)
Comment (natalia-zolochevska) [inline script/controller/tasklist-front.js:16]: and selected should a variable, not a property of item

File: views/tasks-widget.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<div class="widget" ng-controller="widgetController">
+
+    <form name="newTaskForm" novalidate ng-submit="addTask();">
+        <div class="new-task">
+            <textarea required ng-model='description' name="description" class="task-txt new-task-txt" rows=1
+                      placeholder="Enter your task description"></textarea>
+
+            <input ng-model='assignee' class='assign inline-edit' placeholder="Assign to">
```
Comment (natalia-zolochevska) [inline views/tasks-widget.html:9]: please implement some custom validator also
for example, check that user is one of the users you have in db (in array)

File: script/service/tasklist-back.js [hunk 4]
```diff
@@ -63,7 +61,15 @@ angular.module('tasklist-back', ['utils']).
                 return ret;
             };
```
Comment (DmytroArseniuk) [inline script/service/tasklist-back.js:64]: I think you should split methods that belong to task (changeTaskStatus, changeTaskDescription, etc.) and methods that belong to tasklist(add, remove, find). To make you API more clear and structured.

File: script/service/tasklist-back.js [hunk 5]
```diff
@@ -130,7 +134,7 @@ angular.module('tasklist-back', ['utils']).
                 };
             };
 
-            __proto__.restoreFrom = function (data) {
+            TaskItem.prototype.restoreFrom = function (data) {
```
Comment (DmytroArseniuk) [inline script/service/tasklist-back.js:138]: Avoid accessors in angular for data binding.

**Pull request closed without merging**